Sources/SpeziFHIR/FHIRStore.swift

```diff
@@ -19,149 +17,146 @@ import Spezi
 /// The ``FHIRStore`` is automatically injected in the environment if you use the ``FHIR`` standard or can be used as a standalone module.
 @Observable
 public final class FHIRStore: Module,
-                              EnvironmentAccessible,
-                              DefaultInitializable,
-                              @unchecked Sendable /* `unchecked` `Sendable` conformance fine as access to `_resources` protected by `NSLock` */ {
-    private let lock = NSLock()
-    @ObservationIgnored private var _resources: [FHIRResource]
-    
-    
-    /// Allergy intolerances.
+                             EnvironmentAccessible,
+                             DefaultInitializable,
+                             Sendable {
+    /// Actor-isolated storage for `FHIRResource`s backing the ``FHIRStore``.
+    private actor Storage {
+        // Non-isolation required so that resource access via the `FHIRStore` stays sync (required for seamless SwiftUI access).
+        // Isolation is still guaranteed as the only modifying functions `insert()` and `remove()` are isolated on the `FHIRStore.Storage` actor.
+        nonisolated(unsafe) private var _resources: [FHIRResource] = []
+
+
+        var resourceIds: [FHIRResource.ID] {
+            _resources.map(\.id)
+        }
+
+
+        func insert(resource: FHIRResource) {
+            _resources.append(resource)
+        }
+
+        func remove(resource resourceId: FHIRResource.ID) {
+            _resources.removeAll { $0.id == resourceId }
+        }
+
+        nonisolated func fetch(for category: FHIRResource.FHIRResourceCategory) -> [FHIRResource] {
+            _resources.filter { $0.category == category }
+        }
+
+
+        subscript(id: FHIRResource.ID) -> FHIRResource? {
+            _resources.first { $0.id == id }
+        }
+    }
+
+
+    private let storage: Storage
+
+
+    /// `FHIRResource`s with category `allergyIntolerance`.
     public var allergyIntolerances: [FHIRResource] {
         access(keyPath: \.allergyIntolerances)
-        return lock.withLock {
-            _resources.filter { $0.category == .allergyIntolerance }
-        }
+        return storage.fetch(for: .allergyIntolerance)
     }
-    
-    /// Conditions.
+
+    /// `FHIRResource`s with category `condition`.
```

Comment:
What if we can use property wrappers here? Because there are some repetitions in place. For example, accessing the keyPath, fetching the resource.

```swift 
@SomeCustomPropertyWrapper(category: . condition)
public var conditions: [FHIRResource]

@SomeCustomPropertyWrapper(category: . allergyIntolerance)
public var allergyIntolerances: [FHIRResource]
```

Reply:
I would rather keep the number of property wrappers that we introduce low, especially as your suggested wrapper doesn't really add any benefit.

Current syntax:
```swift
struct someView: View {
    @Environment(FHIRStore.self) private var fhirStore

    var body: some View { 
        Text("\(fhirStore.allergyIntolerances.count)")
    }
}
```

Your suggested property wrapper (with the precondition that the `FHIRStore` itself is already in the environment):
```swift
struct someView: View {
    @SomeCustomPropertyWrapper(category: . allergyIntolerance)
    public var allergyIntolerances: [FHIRResource]

    var body: some View { 
        Text("\(allergyIntolerances.count)")
    }
}
```

Reply:
Maybe you misunderstood my suggestion. I was thinking going from 


```swift 
public var allergyIntolerances: [FHIRResource] {
    access(keyPath: \.allergyIntolerances)
    return storage.fetch(for: .allergyIntolerance)
}

public var conditions: [FHIRResource] {
    access(keyPath: \.conditions)
    return storage.fetch(for: .condition)
}

public var diagnostics: [FHIRResource] {
    access(keyPath: \.diagnostics)
    return storage.fetch(for: .diagnostic)
}

...
```

to 

```swift 
@SomeCustomPropertyWrapper(category: .allergyIntolerance)
public var allergyIntolerances: [FHIRResource]

@SomeCustomPropertyWrapper(category: .condition)
public var conditions: [FHIRResource]
    
@SomeCustomPropertyWrapper(category: .diagnostic)
public var diagnostics: [FHIRResource]
     
...
```

Reply:
I see, thanks for clarifying!
However, I feel even stronger against implementing an internal property wrapper that just abstracts a tiny internal code piece, it's simply not worth it.